output/cloud/metrics_client.go

```diff
@@ -64,7 +65,10 @@ func (mc *MetricsClient) PushMetric(referenceID string, s []*Sample) error {
 	var additionalFields logrus.Fields
 
 	if !mc.noCompress {
-		buf := mc.pushBufferPool.Get().(*bytes.Buffer)
+		buf, ok := mc.pushBufferPool.Get().(*bytes.Buffer)
+		if !ok {
+			return errors.New("failed to convert a buffer pool item into the expected type bytes Buffer")
```

Comment:
It can't happen but more important we are not going to use this metric_client in the v2, so I would not spend time on improving this.

I did it only to fix the linter otherwise we will get an error in the new PR moving it to a new file under `v1` pkg.

Reply:
Anyway, I applied a small change.

Reply:
Done, check 8dba987439d8bac8312d17a0f561cb566234399b